core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -61,7 +72,7 @@ private[spark] class CoarseGrainedExecutorBackend(
       // This is a very fast action so we can use "ThreadUtils.sameThread"
       driver = Some(ref)
       ref.ask[Boolean](RegisterExecutor(executorId, self, hostname, cores, extractLogUrls,
-        extractAttributes))
+        extractAttributes, parseOrFindResources(resourcesFile)))
```

Comment:
the comment above about this being "a very fast action" might not be so accurate anymore, now that you're calling a script.  I'm not sure how much that matters.  You could also just do it before the call to `rpcEnv.asyncSetupEndpointRefByURI` to avoid any issues.

Reply:
yeah will move it up